websocket/connection.go

```diff
@@ -27,17 +29,25 @@ const (
 // This is still used by access carrier
 type GorillaConn struct {
 	*websocket.Conn
-	log *zerolog.Logger
+	log     *zerolog.Logger
+	readBuf bytes.Buffer
 }
 
 // Read will read messages from the websocket connection
 func (c *GorillaConn) Read(p []byte) (int, error) {
+	// Intermediate buffer may contain unread bytes from the last read, start there before blocking on a new frame
+	if c.readBuf.Len() > 0 {
+		return c.readBuf.Read(p)
+	}
+
 	_, message, err := c.Conn.ReadMessage()
 	if err != nil {
 		return 0, err
 	}
-	return copy(p, message), nil
-
+	// Write into an intermediate buffer in case the websocket frame contains more data then len(p)
+	// Write returns nil error always and grows the buffer everything is always written or panic
+	c.readBuf.Write(message)
+	return c.readBuf.Read(p)
```

Comment:
Can we avoid the double copy in case we know it is not needed? (i.e., if `len(message) <= len(p)`)

Reply:
just a philosphy I have to not fork the code unless performance is critical in order to not create a 'rarely used' path where bugs can develop unnoticed

Just sharing 'why', but I made the change :) there is now a fast path